lib/Sema/TypeCheckProtocol.cpp

```diff
@@ -1748,6 +1748,81 @@ checkWitness(AccessScope requiredAccessScope,
   return CheckKind::Success;
 }
 
+/// Given a subscript defined as "subscript(dynamicMember:)->T", return true if
+/// it is an acceptable implementation of the DynamicMemberLookupProtocol
+/// requirement.
+bool swift::isAcceptableDynamicMemberLookupSubscript(ValueDecl *decl,
+                                                     DeclContext *DC,
+                                                     TypeChecker &TC) {
+  // Check the conforming decl.
+  auto match = cast<SubscriptDecl>(decl);
```

Comment:
Since you're casting unconditionally it makes more sense to just change the parameter type

Reply:
Yes, good point, fixed.